### PR TITLE
Add PSRAM module to onju-voice.yaml

### PR DESCRIPTION
**Pull request by vhsdream**
Streaming audio seems to work better with this enabled. Also fixed a typo with the friendly_name variable.

### CODE REVIEW

File: esphome/onju-voice.yaml [hunk 1]
```diff
@@ -6,15 +6,17 @@ substitutions:
 
 esphome:
   name: "${name}"
-  friendly_name: "{$friendly_name}"
+  friendly_name: "${friendly_name}"
   comment: "${device_description}"
```
Comment (tetele) [inline esphome/onju-voice.yaml:10]: Thank you for catching this, but please let's have PRs dedicated to one purpose. Can you please remove the change from here? You can contribute another PR for this change specifically or i can make the change myself later today.

File: esphome/onju-voice.yaml [hunk 2]
```diff
@@ -6,15 +6,17 @@ substitutions:
...
   name_add_mac_suffix: true
   project:
     name: tetele.onju_voice_satellite
     version: "${project_version}"
-  min_version: 2023.11.6
+  min_version: 2024.3.0
```
Comment (tetele) [inline esphome/onju-voice.yaml:15]: What is that version required for?

**Review by tetele — CHANGES_REQUESTED** [review]
Thanks a lot for adding this!

**Comment by jherby2k** [comment]
Ive tested this change and can confirm at least with 2024.4 that my audio hitching issue is completely gone while streaming an MP3.

**Comment by vhsdream** [comment]
OK I think I managed to figure it out (I'm not super git-savvy). Will make a PR for the typo fix soon.

**Review by tetele — APPROVED** [review]

**Comment by tetele** [comment]
Thank you!

File: esphome/onju-voice.yaml [hunk 3]
```diff
@@ -15,6 +15,8 @@ esphome:
   min_version: 2023.11.6
   platformio_options:
     board_build.flash_mode: dio
+    build_flags: "-DBOARD_HAS_PSRAM"
```
Comment (jesserockz) [inline esphome/onju-voice.yaml:18]: This flag is set automatically when you include the `psram` component in yaml

**Pull request merged**